web/docs/operators/transformations/flatten.md

```diff
@@ -10,23 +10,27 @@ flatten [<separator>]
 
 ## Description
 
-The `flatten` operator removes any nested lists or records by merging lists and
-joining nested records with a separator. Flattening removes null values.
+The `flatten` operator acts on [container
+types](../../data-model/type-system.md):
 
-:::info
-Unlike for most data models, flattening is an (almost) free operation in VAST's
-data model.
-:::
+1. **Records**: join nested records with a separator. For example, if a field
+   named `x` is a nested records with fields `a` and `b`, flattening will lift
+   the nested records into the parent scope by creating two new fields `x.a` and
+   `x.b`.
+2. **Lists**: merge nested lists into a single (flat) list. For example,
+   `GENERIC-EXAMPLE-MISSING`.
```

Comment:
@Dakostu or @dominiklohmann, please explain to the user how list flattening works.

Reply:
Nested lists will be flattened into a singular list, in the same order of the innermost list elements.

Example:
`{"foo": [[[2]],[[3,1]],[[4]]]}` -> `{"foo": [2,3,1,4]}`

For each record inside a list, the operator will flatten its values into a list in a singular record, in the same order of the record's elements. 

`{"foo": [{"a": 2, "b": 1}, {"a": 4}]}` -> `{"foo.a": [2, 4], "foo.b": [1, null]}`

Nested lists inside records will also be flattened.

`{"foo": [{"a": [[2, 23], [1,16]], "b": [1]}, {"a": [[4]]}]}` -> `{"foo.a": [2, 23, 1, 16, 4], "foo.b": [1]}`

Reply:
Thanks, @Dakostu. I've added those examples to the docs.